core/src/main/java/org/apache/calcite/rel/RelNode.java

```diff
@@ -232,6 +233,21 @@
    */
   RelNode onRegister(RelOptPlanner planner);
 
+  /**
+   * Returns a relational expression string of this {@code RelNode}.
+   * The string returned is the same as
+   * {@link RelOptUtil#toString(org.apache.calcite.rel.RelNode)}.
+   *
+   * This method is intended mainly for use while debugging in an IDE,
+   * as a convenient short-hand for RelOptUtil.toString.
+   * We recommend that sub-classes do not override this method.
+   *
+   * @return Relational expression string of this {@code RelNode}
+   */
+  default String explain() {
```

Comment:
nitpicking: I think this new method should go together (e.g. right after) with `void explain(RelWriter pw);`
Also, since `RelNode` is an interface (and not an abstract class), IMHO javadoc should not say "sub-classes", but instead something along the lines: "We recommend that classes implementing this interface do not override this method"

Reply:
@rubenada Thanks for your advice, indeed better!

Reply:
revised